temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -230,62 +302,36 @@ private Result handleWorkflowTaskWithEmbeddedQuery(
       }
       throw e;
     } finally {
-      if (stickyTaskQueueName == null && workflowRunTaskHandler != null) {
-        log.trace(
-            "Closing runner for {}-{} because of non-sticky worker configuration",
-            execution.getWorkflowId(),
-            execution.getRunId());
-        workflowRunTaskHandler.close();
-      } else {
+      if (useCache) {
         cache.markProcessingDone(execution);
+      } else if (workflowRunTaskHandler != null) {
+        workflowRunTaskHandler.close();
```

Comment:
Should this be just `if` rather than `else if`? If we're not using the cache will the handler just be closed because it's disposed of?

Reply:
I had this same question: https://github.com/temporalio/sdk-java/pull/881#discussion_r751277467. Apparently this is by intention.